### PR TITLE
571 rewrite buttons to be wrappers around blueprintjss buttons

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/zakodium-oss/react-science/issues/571

### CODE REVIEW

File: src/components/button/Button.tsx [hunk 1]
```diff
@@ -1,66 +1,31 @@
...
-        },
-      })}
-      type="button"
-      disabled={disabled}
-    >
-      {children}
-    </button>
+    <InnerButton {...buttonProps}>
+      <Tooltip {...tooltipProps}>{children}</Tooltip>
+    </InnerButton>
```
Comment (stropitek) [inline src/components/button/Button.tsx:29]: Check that the tooltip is really activated on the whole button, also when bringing the cursor close to the edges.

**Review by stropitek — CHANGES_REQUESTED** [review]
CleanShot 2023-11-02 at 11 30 17

Can you add some padding so that we can see the border of the button better in the button stories?

---

Can you add a story which is button with a tooltip in which the tooltip props are controllable?

---

I think the disabled story can be removed since that can be controlled in the main controllable story.

---

There is a regression in the toolbar buttons: https://react-science.pages.dev/stories/?path=/story/components-aboutdialogtoolbarbutton--display-about-dialog

On main: 
CleanShot 2023-11-02 at 11 32 37

On this PR: 
CleanShot 2023-11-02 at 11 32 47

---

There is also a regression here (PanelPreferencesToolbar story). I think the minimal version of the button with the icon prop (use blueprint icons) should work no?

Other regression: PanelHeader

---

Can the close button of the modal also be written with the button component? Again always try to use icons from the bluprintjs library first.

File: src/components/button/Button.tsx (hunk 1 above)
Comment (wadjih-bencheikh18) [reply src/components/button/Button.tsx:29]: resolved

**Comment by wadjih-bencheikh18** [comment]
> use blueprint icons

I used blueprint icons in all cases, but we can see that these icons are grey
Do I have to change the color or keep it like that (I prefer the actual colors)

File: src/components/button/Button.tsx [hunk 2]
```diff
@@ -1,66 +1,31 @@
 /** @jsxImportSource @emotion/react */
```
Comment (stropitek) [inline src/components/button/Button.tsx:1]: This line can be removed

**Review by stropitek — APPROVED** [review]
Can you check why the `fill` prop has no effect in the Button story?

It's supposed to be working like here: https://blueprintjs.com/docs/#core/components/buttons

Otherwise LGTM

**Comment by wadjih-bencheikh18** [comment]
> Can you check why the `fill` prop has no effect in the Button story?

The problem is happening because `button` is wrapped inside the `tooltip` so he's filling the tooltip.
But I fixed it.

**Pull request merged**